sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/DoubleExponentialHistogramBuckets.java

```diff
@@ -234,19 +257,44 @@ public boolean equals(@Nullable Object obj) {
     DoubleExponentialHistogramBuckets other = (DoubleExponentialHistogramBuckets) obj;
     // Don't need to compare getTotalCount() because equivalent bucket counts
     // imply equivalent overall count.
-    return getBucketCounts().equals(other.getBucketCounts())
-        && this.getOffset() == other.getOffset()
-        && this.scale == other.scale;
+    return this.scale == other.scale && sameBucketCounts(other);
+  }
+
+  /**
+   * Tests if two bucket counts are equivalent semantically.
+   *
+   * <p>Semantic equivalence means:
+   *
+   * <ul>
+   *   <li>All counts are stored between indexStart/indexEnd.
+   *   <li>Offset does NOT need to be the same
+   * </ul>
+   */
+  private boolean sameBucketCounts(DoubleExponentialHistogramBuckets other) {
+    int min = Math.min(this.counts.getIndexStart(), other.counts.getIndexStart());
+    int max = Math.max(this.counts.getIndexEnd(), other.counts.getIndexStart());
```

Comment:
Should this be referencing other.getIndexEnd()?

Reply:
ouch, good catch.   I got lucky in the tests based on how it orders equals, will fix.

Reply:
fixed

Reply:
Can we add the unlucky unit test for this?

Reply:
I'd rather find a way to make the unit tests NOT care about equality here, but yeah I'll see if I can get a better unit test (not see my comment to james on alternatives here).

Reply:
Updated the unit test.  I think we may be able to do some refactoring of ExponentialHistogramBuckets at some point to clean up this code, and hopefully ensure we're not relying on hashCode/equals in production.